x/evm/keeper/evm.go

```diff
@@ -393,21 +414,28 @@ func (k *Keeper) ApplyMessageWithConfig(ctx cosmos.Context,
 		// - reset sender's nonce to msg.Nonce() before calling evm.
 		// - increase sender's nonce by one no matter the result.
 		stateDB.SetNonce(sender.Address(), msg.Nonce)
-		ret, _, leftoverGas, vmErr = evm.Create(sender, msg.Data, leftoverGas, msg.Value)
+		ret, _, leftoverGas, vmErr = evm.Create(aspectCtx, sender, msg.Data, leftoverGas, msg.Value)
 		stateDB.SetNonce(sender.Address(), msg.Nonce+1)
 	} else {
 		pointRequest := k.aspect.CreateTxPointRequestInEvm(ctx, msg, txConfig, nil)
 		pointRequest.GasInfo.Gas = leftoverGas
 
-		execute := djpm.AspectInstance().PreTxExecute(pointRequest)
+		execute := djpm.AspectInstance().PreTxExecute(aspectCtx, pointRequest)
 		leftoverGas = execute.GasInfo.Gas
 		if hasErr, execErr := execute.HasErr(); hasErr {
 			vmErr = execErr
 		} else {
-			ret, leftoverGas, vmErr = evm.Call(sender, *msg.To, msg.Data, leftoverGas, msg.Value)
+			// When handling joinpoints such as PreContractCall and PostContractCall, it is possible that a JIT (Just-In-Time) call may occur.
+			// In such cases, it is essential to set the jit manager for the EVM call.
+			// TODO: It is important to note that the manager of the JIT should not rely on a global instance.
+			// TODO： Consider revising this implementation for better design and modularity.
+			newAspectProtocol := provider.NewAspectProtocolProvider(aspectCtx.EthTxContext)
+			jit_inherent.Update(newAspectProtocol)
```

Comment:
Will this cause some race conditions when doing `eth_call` ? Because we are updating the global variable here.

Reply:
@dumbeng Refactored the call to the JIT manager by eliminating the global instance mode.